### PR TITLE
fix(Divider): Fix rtl styles

### LINKED ISSUE #13337: Divider: No spaces between the lines and content in RTL context
### Environment Information

- **Package version(s)**: 0.49.0
- **Browser and OS versions**: Not relevant

#### Sandbox: 
https://codesandbox.io/s/divider-content-rtl-jzwql
#### Screenshot:
Divider with RTL content issue

#### Actual behavior:
No spaces between the lines and the content

#### Expected behavior:
There should be spaces between the lines and the content (as the same of LTR context)

### Priorities and help requested:

Are you willing to submit a PR to fix? **Yes**

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #13337
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Fix RTL styles for `Divider` in RTL context

Screenshot 2020-06-23 at 07 44 59

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts [hunk 1]
```diff
@@ -40,11 +40,13 @@ const dividerStyles: ComponentSlotStylesPrepared<DividerStylesProps, DividerVari
             lineHeight: variables.textLineHeight,
             '::before': {
               ...beforeAndAfter(size, variables, colors, props),
-              marginRight: pxToRem(20),
+              ...(!rtl && { marginRight: pxToRem(20) }),
+              marginInlineEnd: pxToRem(20),
```
Comment (miroslavstastny) [inline packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts:44]: `margin-inline-end` is [not supported in IE11](https://caniuse.com/#search=margin-inline-end).

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts:44]: The alternative could be create a slot for the text and apply the margins on it, but @layershifter mentioned it can be too much to solve this...

Initially I didn't saw a problem in the related issue itself, the RTL text works in RTL context and LTR in the LTR... But if we are going to solve it what can be the best solution then?

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts]: I pushed new code solving it with a slot.

File: packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ const beforeAndAfter = (
 });
 
 const dividerStyles: ComponentSlotStylesPrepared<DividerStylesProps, DividerVariables> = {
-  root: ({ props, variables }): ICSSInJSStyle => {
+  root: ({ props, variables, rtl }): ICSSInJSStyle => {
```
Comment (assuncaocharles) [inline packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts:22]:
```suggestion
  root: ({ props, variables }): ICSSInJSStyle => {
```

**Comment by assuncaocharles** [comment]
> Please add screener example to catch regressions.

Screener added

File: packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts (hunk 1 above)
Comment (abdullahsalem) [reply packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts]: You are right @assuncaocharles, initially it looks like there is no an issue (when the direction of the language meets the direction of the context flow), but in some cases there may be a need to fill that content with a language that has a different direction than the context flow direction. 
BTW, I've mentioned the main reason that leads to this issue, [here](https://github.com/microsoft/fluentui/issues/13337#issuecomment-634954807).

In other words:
> The `auto` value tells the browser to look at the first strongly typed character in the element and work out from that what the base direction of the element should be.
[w3.org](https://www.w3.org/International/questions/qa-html-dir#dirauto)

Therefore, I would suggest removing that `auto` if there is no any other side effect should be considered.

File: packages/fluentui/docs/src/examples/components/Divider/Visuals/DividerExampleRtlAndLtr.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Divider, Provider, Card, CardHeader, CardBody, Text } from '@fluentui/react-northstar';
+
+const DividerExampleFittedShorthand = () => (
+  <>
+    <Provider>
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Divider/Visuals/DividerExampleRtlAndLtr.tsx:6]:
```suggestion
```

File: packages/fluentui/docs/src/examples/components/Divider/Visuals/DividerExampleRtlAndLtr.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Divider, Provider, Card, CardHeader, CardBody, Text } from '@fluentui/react-northstar';
+
+const DividerExampleFittedShorthand = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Divider/Visuals/DividerExampleRtlAndLtr.tsx:4]:
```suggestion
const DividerExampleRtlAndLtr = () => (
```

File: packages/fluentui/docs/src/examples/components/Divider/Visuals/index.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+
+import ComponentExample from '../../../../components/ComponentDoc/ComponentExample';
+import NonPublicSection from '../../../../components/ComponentDoc/NonPublicSection';
+
+const UsageVisual = () => (
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Divider/Visuals/index.tsx:6]:
```suggestion
const Visual = () => (
```

File: packages/fluentui/react-northstar/src/components/Divider/Divider.tsx [hunk 6]
```diff
@@ -91,11 +95,17 @@ const Divider: React.FC<WithAsProp<DividerProps>> & FluentComponentStaticProps<D
     rtl: context.rtl,
   });
 
+  const content = createShorthand(DividerContent, props.content, {
+    defaultProps: () => ({
+      styles: resolvedStyle.content,
+    }),
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/components/Divider/Divider.tsx:101]:
```suggestion
```

Let's move styles to `dividerContentStyles`

File: packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts [hunk 7]
```diff
@@ -54,6 +52,10 @@ const dividerStyles: ComponentSlotStylesPrepared<DividerStylesProps, DividerVari
           }),
     };
   },
+  content: () => ({
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/themes/teams/components/Divider/dividerStyles.ts:55]: Please move this to `dividerContentStyles`
